<!DOCTYPE html>
<html lang='en'>
<head>
<title>0001-ozone-wayland-implement-text_input_manager-fixes.patch - aur.git - AUR Package Repositories</title>
<meta name='generator' content='cgit v1.2.3-56-gcc08'/>
<meta name='robots' content='index, nofollow'/>
<link rel='stylesheet' type='text/css' href='/static/css/cgit.css'/>
<script type='text/javascript' src='/cgit.js'></script>
<link rel='shortcut icon' href='/static/images/favicon.ico'/>
<link rel='alternate' title='Atom feed' href='https://aur.archlinux.org/cgit/aur.git/atom/0001-ozone-wayland-implement-text_input_manager-fixes.patch?h=chromium-wayland-vaapi' type='application/atom+xml'/>
<link rel='vcs-git' href='https://aur.archlinux.org/chromium-wayland-vaapi.git/' title='aur.git Git repository'/>
</head>
<body>
		<div id="archnavbar" class="anb-aur">
			<div id="archnavbarlogo"><h1><a href="/" title="Return to the main page">Arch Linux User Repository</a></h1></div>
			<div id="archnavbarmenu">
				<ul id="archnavbarlist">
					<li id="anb-home"><a href="https://www.archlinux.org/" title="Arch news, packages, projects and more">Home</a></li>
					<li id="anb-packages"><a href="https://www.archlinux.org/packages/" title="Arch Package Database">Packages</a></li>
					<li id="anb-forums"><a href="https://bbs.archlinux.org/" title="Community forums">Forums</a></li>
					<li id="anb-wiki"><a href="https://wiki.archlinux.org/" title="Community documentation">Wiki</a></li>
					<li id="anb-bugs"><a href="https://bugs.archlinux.org/" title="Report and track bugs">Bugs</a></li>
					<li id="anb-security"><a href="https://security.archlinux.org/" title="Arch Linux Security Tracker">Security</a></li>
					<li id="anb-aur"><a href="/" title="Arch Linux User Repository">AUR</a></li>
					<li id="anb-download"><a href="https://www.archlinux.org/download/" title="Get Arch Linux">Download</a></li>
				</ul>
			</div>
		</div><!-- #archnavbar -->
<div id='cgit'><table id='header'>
<tr>
<td class='main'>AUR : chromium-wayland-vaapi.git</td></tr>
<tr><td class='sub'>AUR Package Repositories | click <a href='/pkgbase/chromium-wayland-vaapi/'>here</a> to return to the package base details page</td><td class='sub right'></td></tr></table>
<table class='tabs'><tr><td>
<a href='/cgit/aur.git/?h=chromium-wayland-vaapi'>summary</a><a href='/cgit/aur.git/log/0001-ozone-wayland-implement-text_input_manager-fixes.patch?h=chromium-wayland-vaapi'>log</a><a class='active' href='/cgit/aur.git/tree/0001-ozone-wayland-implement-text_input_manager-fixes.patch?h=chromium-wayland-vaapi'>tree</a><a href='/cgit/aur.git/commit/0001-ozone-wayland-implement-text_input_manager-fixes.patch?h=chromium-wayland-vaapi'>commit</a><a href='/cgit/aur.git/diff/0001-ozone-wayland-implement-text_input_manager-fixes.patch?h=chromium-wayland-vaapi'>diff</a><a href='/cgit/aur.git/stats/0001-ozone-wayland-implement-text_input_manager-fixes.patch?h=chromium-wayland-vaapi'>stats</a></td><td class='form'><form class='right' method='get' action='/cgit/aur.git/log/0001-ozone-wayland-implement-text_input_manager-fixes.patch'>
<input type='hidden' name='h' value='chromium-wayland-vaapi'/><select name='qt'>
<option value='grep'>log msg</option>
<option value='author'>author</option>
<option value='committer'>committer</option>
<option value='range'>range</option>
</select>
<input class='txt' type='search' size='10' name='q' value=''/>
<input type='submit' value='search'/>
</form>
</td></tr></table>
<div class='path'>path: <a href='/cgit/aur.git/tree/?h=chromium-wayland-vaapi'>root</a>/<a href='/cgit/aur.git/tree/0001-ozone-wayland-implement-text_input_manager-fixes.patch?h=chromium-wayland-vaapi'>0001-ozone-wayland-implement-text_input_manager-fixes.patch</a></div><div class='content'>blob: 0d4e551956cf4f568897f2bf28d504552c3ba2b3 (<a href='/cgit/aur.git/plain/0001-ozone-wayland-implement-text_input_manager-fixes.patch?h=chromium-wayland-vaapi'>plain</a>)
<table summary='blob content' class='blob'>
<tr><td class='linenumbers'><pre><a id='n1' href='#n1'>1</a>
<a id='n2' href='#n2'>2</a>
<a id='n3' href='#n3'>3</a>
<a id='n4' href='#n4'>4</a>
<a id='n5' href='#n5'>5</a>
<a id='n6' href='#n6'>6</a>
<a id='n7' href='#n7'>7</a>
<a id='n8' href='#n8'>8</a>
<a id='n9' href='#n9'>9</a>
<a id='n10' href='#n10'>10</a>
<a id='n11' href='#n11'>11</a>
<a id='n12' href='#n12'>12</a>
<a id='n13' href='#n13'>13</a>
<a id='n14' href='#n14'>14</a>
<a id='n15' href='#n15'>15</a>
<a id='n16' href='#n16'>16</a>
<a id='n17' href='#n17'>17</a>
<a id='n18' href='#n18'>18</a>
<a id='n19' href='#n19'>19</a>
<a id='n20' href='#n20'>20</a>
<a id='n21' href='#n21'>21</a>
<a id='n22' href='#n22'>22</a>
<a id='n23' href='#n23'>23</a>
<a id='n24' href='#n24'>24</a>
<a id='n25' href='#n25'>25</a>
<a id='n26' href='#n26'>26</a>
<a id='n27' href='#n27'>27</a>
<a id='n28' href='#n28'>28</a>
<a id='n29' href='#n29'>29</a>
<a id='n30' href='#n30'>30</a>
<a id='n31' href='#n31'>31</a>
<a id='n32' href='#n32'>32</a>
<a id='n33' href='#n33'>33</a>
<a id='n34' href='#n34'>34</a>
<a id='n35' href='#n35'>35</a>
<a id='n36' href='#n36'>36</a>
<a id='n37' href='#n37'>37</a>
<a id='n38' href='#n38'>38</a>
<a id='n39' href='#n39'>39</a>
<a id='n40' href='#n40'>40</a>
<a id='n41' href='#n41'>41</a>
<a id='n42' href='#n42'>42</a>
</pre></td>
<td class='lines'><pre><code>
ui/ozone/platform/wayland/host/zwp_text_input_wrapper.h

# https://github.com/chromium/chromium/commit/de30ba3f21f824be05443d5820b988d226780d68
Support offset of surrounding text in Lacros wayland client.

# https://github.com/chromium/chromium/commit/71bba2b7ae8117fdf053563a864d6cff018d7e94
Support large size surrounding text in Lacros.

<span class="hl kwa">---</span>
diff --git a/ui/ozone/platform/wayland/host/zwp_text_input_wrapper_v3.cc b/ui/ozone/platform/wayland/host/zwp_text_input_wrapper_v3.cc
index a3ce6e4..83f2c58 100644
<span class="hl kwa">--- a/ui/ozone/platform/wayland/host/zwp_text_input_wrapper_v3.cc</span>
<span class="hl kwb">+++ b/ui/ozone/platform/wayland/host/zwp_text_input_wrapper_v3.cc</span>
&#64;&#64; -147,6 +147,14 &#64;&#64; void ZWPTextInputWrapperV3::SetSurroundingText(
   zwp_text_input_v3_commit(obj_.get());
 }
 
<span class="hl kwb">+bool ZWPTextInputWrapperV3::HasAdvancedSurroundingTextSupport() const {</span>
<span class="hl kwb">+	return false;</span>
<span class="hl kwb">+}</span>
<span class="hl kwb">+</span>
<span class="hl kwb">+void ZWPTextInputWrapperV3::SetSurroundingTextOffsetUtf16(</span>
<span class="hl kwb">+    uint32_t offset_utf16) {</span>
<span class="hl kwb">+}</span>
<span class="hl kwb">+</span>
 void ZWPTextInputWrapperV3::ResetPendingState() {
   commit_string_.clear();
   delete_surrounding_text_before_length_ = 0;
diff --git a/ui/ozone/platform/wayland/host/zwp_text_input_wrapper_v3.h b/ui/ozone/platform/wayland/host/zwp_text_input_wrapper_v3.h
index 204d7e3..5d03a1d 100644
<span class="hl kwa">--- a/ui/ozone/platform/wayland/host/zwp_text_input_wrapper_v3.h</span>
<span class="hl kwb">+++ b/ui/ozone/platform/wayland/host/zwp_text_input_wrapper_v3.h</span>
&#64;&#64; -45,6 +45,8 &#64;&#64; class ZWPTextInputWrapperV3 : public ZWPTextInputWrapper {
   void SetCursorRect(const gfx::Rect&amp; rect) override;
   void SetSurroundingText(const std::string&amp; text,
                           const gfx::Range&amp; selection_range) override;
<span class="hl kwb">+  bool HasAdvancedSurroundingTextSupport() const override;</span>
<span class="hl kwb">+  void SetSurroundingTextOffsetUtf16(uint32_t offset_utf16) override;</span>
   void SetContentType(TextInputType type,
                       TextInputMode mode,
                       uint32_t flags,
</code></pre></td></tr></table>
</div> <!-- class=content -->
<div id="footer">
	<p>
		Copyright &copy; 2004-2023 aurweb Development Team &ndash;
		<strong>AUR packages are user produced content. Any use of the provided files is at your own risk.</strong>
	</p>
</div>
</div> <!-- id=cgit -->
</body>
</html>
